server/server_event.py

```diff
@@ -81,9 +85,17 @@ async def execute_action(self, game: Game):
             game.game_id,
             self.response
         )
-        game.next_turn()
-        data_received.turn_data.update({'turn_token': game.turn_token})
-        return await notify_your_turn(
-            data_received.current_player,
-            data_received.turn_data,
-        )
+        if data_received.current_player == '':
+            game.state = 'ended'
```

Comment:
You should move this a constant and improve that validation. Something like

```
ENDED = 'ended'

if not data_received.current_player:
    game.state = ENDED
```

Reply:
I have a question, are you gonna store those states in the code or elsewhere like a DB? In that case, using string as state definitions isn't the best approach in terms of performance

Reply:
Changed to constants that have int values.